src/main/scala/tech/cryptonomic/conseil/tezos/TezosNodeOperator.scala

```diff
@@ -325,19 +326,24 @@ class TezosNodeOperator(val node: TezosRPCInterface, val network: String, batchC
   /**
     * Gets last `depth` blocks.
     * @param depth      Number of latest block to fetch, `None` to get all
+    * @param headHash   Hash of a block from which to start, None to start from a real head
     * @return           Blocks and Account hashes involved
     */
-  def getLatestBlocks(depth: Option[Int] = None): Future[PaginatedBlocksResults] =
-    getBlockHead().map {
-      head =>
-        val headLevel = head.data.header.level
-        val headHash = head.data.hash
-        val minLevel = depth.fold(1)(d => max(1, headLevel - d + 1))
-        val pagedResults = partitionBlocksRanges(minLevel to headLevel).map(
-          page => getBlocks((headHash, headLevel), page)
-        )
-        (pagedResults, headLevel - minLevel + 1)
-    }
+  def getLatestBlocks(depth: Option[Int] = None, headHash: Option[String] = None): Future[PaginatedBlocksResults] = {
+    headHash
+      .map(it => getBlock(BlockHash(it)))
+      .getOrElse(getBlockHead())
+      .map {
+        head =>
```

Comment:
This should better be renamed, to avoid confusion.
Something like `reference`, `max`, `from`, ...

Reply:
done